tests/integration/tests/services_test.go

```diff
@@ -0,0 +1,156 @@
+	}
+	assert.NotNil(serviceList.Validations)
+	assert.Equal(utils.BOOKINFO, serviceList.Namespace.Name)
+}
+
+func TestServiceDetails(t *testing.T) {
+	name := "productpage"
+	assert := assert.New(t)
+	service, err := utils.ServiceDetails(name, utils.BOOKINFO)
+
+	assert.Nil(err)
+	assert.NotNil(service)
+	assert.NotNil(service.Service)
+	assert.Equal(utils.BOOKINFO, service.Service.Namespace.Name)
+	assert.NotEmpty(service.Workloads)
+	assert.NotEmpty(service.Service.Ports)
+	assert.NotEmpty(service.Service.Ports)
+	assert.NotNil(service.Endpoints)
+	assert.NotNil(service.VirtualServices)
+	assert.NotNil(service.DestinationRules)
+	assert.NotNil(service.Validations)
+
+	assert.NotNil(service.Health)
+	assert.NotNil(service.Health.Requests)
+	assert.NotNil(service.Health.Requests.Outbound)
+	assert.NotNil(service.Health.Requests.Inbound)
+
+	assert.True(service.IstioSidecar)
+}
+
+func TestServiceDiscoverVS(t *testing.T) {
+	assert := assert.New(t)
+	serviceName := "reviews"
+	vsName := "reviews"
+	service, err := utils.ServiceDetails(serviceName, utils.BOOKINFO)
+	assert.Nil(err)
+	assert.NotNil(service)
+	preVsCount := len(service.VirtualServices)
+	defer utils.OCDelete(utils.VS, utils.BOOKINFO)
+	assert.True(utils.OCApply(utils.VS, utils.BOOKINFO))
+
+	found := false
+	for i := 0; i < 60; i++ {
+		service, err = utils.ServiceDetails(serviceName, utils.BOOKINFO)
+		assert.Nil(err)
+		assert.NotNil(service)
+		if len(service.VirtualServices) > preVsCount {
+			found = true
+			break
+		} else {
+			time.Sleep(time.Second)
+		}
+	}
+	assert.True(found)
+	assert.NotEmpty(service.VirtualServices)
+	found = false
+	for _, vs := range service.VirtualServices {
+		if vs.Name == vsName {
+			found = true
+
+			http := vs.Spec.Http
+			assert.NotEmpty(http)
+			routes := http[0].Route
+			assert.Len(routes, 2)
+
+			assert.Equal(routes[0].Weight, int32(80))
+			destination := routes[0].Destination
+			assert.NotNil(destination)
+			assert.Equal(destination.Host, "reviews")
+			assert.Equal(destination.Subset, "v1")
+
+			assert.Equal(routes[1].Weight, int32(20))
+			destination = routes[1].Destination
+			assert.NotNil(destination)
+			assert.Equal(destination.Host, "reviews")
+			assert.Equal(destination.Subset, "v2")
+
+			break
+		}
+	}
+	assert.True(found)
+}
+
+func TestServiceDiscoverDR(t *testing.T) {
+	assert := assert.New(t)
+	serviceName := "reviews"
+	drName := "reviews"
+	service, err := utils.ServiceDetails(serviceName, utils.BOOKINFO)
+	assert.Nil(err)
+	assert.NotNil(service)
+	preDrCount := len(service.DestinationRules)
+	defer utils.OCDelete(utils.DR, utils.BOOKINFO)
+	assert.True(utils.OCApply(utils.DR, utils.BOOKINFO))
+
+	found := false
+	for i := 0; i < 60; i++ {
+		service, err = utils.ServiceDetails(serviceName, utils.BOOKINFO)
+		assert.Nil(err)
+		assert.NotNil(service)
+		if len(service.DestinationRules) > preDrCount {
+			found = true
+			break
+		} else {
+			time.Sleep(time.Second)
+		}
+	}
```

Comment:
You can use [this helper func](https://pkg.go.dev/k8s.io/apimachinery@v0.23.5/pkg/util/wait#Poll) to poll for a condition if it's useful to you.